online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -196,6 +249,19 @@ class FetcherBase(kvStore: KVStore,
     }
   }
 
+  override def isCachingEnabled(groupBy: GroupBy): Boolean = {
+    if (!isCacheSizeConfigured || groupBy.getMetaData == null || groupBy.getMetaData.getName == null) return false
+
+    val isCachingFlagEnabled = flagStore.isSet("enable_fetcher_batch_ir_cache",
```

Comment:
could the flagstore be null here as the default value? 
Could you add more details about how the flag store information is being created? 

If the flags are getting more and more, we should consider moving all the flags with its functionality to a single file.

Reply:
Good point. I think it could be `null`. 

The FlagStore is passed in via the Chronon API. Users can override with their own implementation. https://github.com/airbnb/chronon/pull/686. I'll add some more information in the flagstore file to clarify.